src/App.js

```diff
@@ -2,34 +2,75 @@ import React from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
-const useSemiPersistentState = () => {
-  const [todoList, setTodoList] = React.useState(
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  );
+// const useSemiPersistentState = () => {
+//   const [todoList, setTodoList] = React.useState(
+//     JSON.parse(localStorage.getItem('savedTodoList')) || []
+//   );
 
-  React.useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-  }, [todoList]);
+//   React.useEffect(() => {
+//     localStorage.setItem('savedTodoList', JSON.stringify(todoList));
+//   }, [todoList]);
 
-  return [todoList, setTodoList];
-};
+//   return [todoList, setTodoList];
+// };
 
 function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
+  const [todoList, setTodoList] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(true);
+
+  React.useEffect(() => {
+    new Promise((resolve, reject) => {
```

Comment:
my only suggestion is to extract the `new Promise` to its own functions to make it cleaner.

Reply:
@Atos20 Noted, thanks for the feedback. I created a new function for this new Promise and pushed it to the remote branch.